scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala

```diff
@@ -648,87 +648,71 @@ class ScalametaParser(input: Input)(implicit dialect: Dialect) { parser =>
    */
   trait PatternContextSensitive {
     private def tupleInfixType(allowFunctionType: Boolean = true): Type = autoPos {
-      require(token.is[LeftParen] && debug(token))
-      val openParenPos = in.tokenPos
-
       // NOTE: This is a really hardcore disambiguation caused by introduction of Type.Method.
       // We need to accept `(T, U) => W`, `(x: T): x.U` and also support unquoting.
       var hasParams = false
       var hasImplicits = false
       var hasTypes = false
-      var secondOpenParenPos = -1
       var closeParenPos = -1
-      val rawtss: List[List[Tree]] = {
-
-        def typedParam() = autoPos {
-          val name = typeName()
-          accept[Colon]
-          Type.TypedParam(name, typ())
-        }
-        @tailrec
-        def paramOrType(): Tree = token match {
-          case Ellipsis(rank) =>
-            ellipsis(rank, astInfo[Tree])
-          case Unquote() =>
-            unquote[Tree]
-          case KwImplicit() if !hasImplicits =>
```

Comment:
it's not directly related to this pr but I got curious
@kitbellew @tgodzik does anybody know in which case there might be `implicit`? :thinking: 

Currently, we allow the following: `type X = (implicit A) => Int` but it's invalid.

Reply:
Not really, this might be accepted by the parser because it was when we forked off from the original parser, but might no longer be relevant.

Reply:
I think it's good as is, no need to remove implicit here.